### DISCUSSION TITLE
help me pls! (in javascript)

### DISCUSSION THREAD

**Discussion by Beluker** [community/community#68824] [Programming Help] [answered]
### Body

i am coding an evolution simulation with neural networks with javascript and am having problems when i apply the kill the creatures not following the selection criteria. I start with 600 creatures in an array. Those who dont meet the criteria die and are removed from the array. So now there might be just 200 left. to simulate reproducing, i need to fill the rest of the array with the remaining survivors, but i am stuck for ideas on how to do this! Help would be much appriciated

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by vishalkrishna8** [comment, marked as answer]
Create offspring by pairing surviving creatures, combining their genes through techniques like crossover and mutation, and then adding the offspring to the population. This replicates genetic evolution and helps your creatures adapt and improve over time.
Hope this helps you!

**Reply by Beluker** [reply]
thank you!
